pkg/certificate/providers/config.go

```diff
@@ -154,6 +154,45 @@ func (c *Config) GetCertificateManager() (certificate.Manager, debugger.Certific
 	}
 }
 
+// SynchronizeCertificate is a helper function that ensures creation and synchronization of a certificate
+// using Kubernetes Secrets backend and API atomicity.
+func SynchronizeCertificate(ns string, secretName string, cert certificate.Certificater, kubeClient kubernetes.Interface) (certificate.Certificater, error) {
```

Comment:
Because we have a few different kinds of certificates in OSM, `SynchronizeCertificate` sounds ambiguous to me.

Should we call this `SynchronizeRootCertificateToKubeSecret` or something, which explicitly tells us what is being done to what?

Syncronization could also be shipping newly renewed certs to Envoys (which I don't think this here is).

Reply:
Or maybe SynchronizeCABundle...

Reply:
it's not going to be for _only_ CA bundle anymore :)
but I agree, @shashankram already suggested a name change